cloud/pkg/router/listener/http.go

```diff
@@ -98,85 +107,132 @@ func (rh *RestHandler) matchedPath(uri string) (string, bool) {
 }
 
 func (rh *RestHandler) httpHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Transfer-Encoding", "chunked")
 	uriSections := strings.Split(r.RequestURI, "/")
 	if len(uriSections) < 2 {
 		// URL format incorrect
-		klog.Warningf("url format incorrect: %s", r.URL.String())
-		w.WriteHeader(http.StatusNotFound)
-		if _, err := w.Write([]byte("Request error")); err != nil {
-			klog.Errorf("Response write error: %s, %s", r.RequestURI, err.Error())
-		}
+		err := fmt.Errorf("url format incorrect: %s", r.URL.String())
+		writeErr(w, r, http.StatusNotFound, err)
 		return
 	}
 
-	matchPath, exist := rh.matchedPath(r.RequestURI)
-	if !exist {
-		klog.Warningf("URL format incorrect: %s", r.RequestURI)
-		w.WriteHeader(http.StatusNotFound)
-		if _, err := w.Write([]byte("Request error")); err != nil {
-			klog.Errorf("Response write error: %s, %s", r.RequestURI, err.Error())
-		}
-		return
-	}
-	v, ok := rh.handlers.Load(matchPath)
-	if !ok {
-		klog.Warningf("No matched handler for path: %s", matchPath)
-		return
-	}
-	handle, ok := v.(Handle)
-	if !ok {
-		klog.Errorf("invalid convert to Handle. match path: %s", matchPath)
-		return
-	}
 	aReaderCloser := http.MaxBytesReader(w, r.Body, MaxMessageBytes)
 	b, err := io.ReadAll(aReaderCloser)
 	if err != nil {
-		klog.Errorf("request error, write result: %v", err)
-		w.WriteHeader(http.StatusBadRequest)
-		if _, err = w.Write([]byte("Request error,body is null")); err != nil {
-			klog.Errorf("Response write error: %s, %s", r.RequestURI, err.Error())
-		}
+		writeErr(w, r, http.StatusBadRequest, err)
 		return
 	}
 
-	if isNodeName(uriSections[1]) {
-		params := make(map[string]interface{})
-		msgID := uuid.New().String()
-		params["messageID"] = msgID
-		params["request"] = r
-		params["timeout"] = rh.restTimeout
-		params["data"] = b
+	edgeNodeName := uriSections[1]
+	err = retry.Do(
```

Comment:
This retries don't seem very useful(only for network fluctuations), it would be strange to use retry only here.